File: backend/src/main/java/turip/favorite/controller/dto/response/FavoriteContentResponse.java
package turip.favorite.controller.dto.response;

import java.time.LocalDate;
import turip.content.controller.dto.response.content.ContentResponse;
import turip.favorite.domain.FavoriteContent;

public record FavoriteContentResponse(
        Long id,
        LocalDate createdAt,
        Long memberId,
        ContentResponse content
) {

    public static FavoriteContentResponse from(FavoriteContent favoriteContent) {
        return new FavoriteContentResponse(
                favoriteContent.getId(),
                favoriteContent.getCreatedAt(),
                favoriteContent.getMember().getId(),
                ContentResponse.of(favoriteContent.getContent(), true)
        );
    }
}

File: android/app/src/main/java/com/on/turip/data/favorite/dto/FavoriteContentResponse.kt
package com.on.turip.data.favorite.dto

import com.on.turip.data.content.dto.ContentResponse
import com.on.turip.data.content.dto.TripDurationInformationResponse
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class FavoriteContentResponse(
    @SerialName("content")
    val content: ContentResponse,
    @SerialName("tripDuration")
    val tripDuration: TripDurationInformationResponse,
    @SerialName("tripPlaceCount")
    val tripPlaceCount: Int,
)
